turbopack/crates/turbopack-core/src/module_graph/chunk_group_info.rs

```diff
@@ -446,205 +442,171 @@ pub async fn compute_chunk_group_info(graph: &ModuleGraph) -> Result<Vc<ChunkGro
+                                {
+                                    // Add bits from parent, and continue traversal because changed
+                                    **current_chunk_groups |= &**parent_chunk_groups;
+                                    GraphTraversalAction::Continue
+                                } else {
+                                    // Unchanged, no need to forward to children
+                                    GraphTraversalAction::Skip
+                                }
+                            }
+                        }
                     }
-                }
-            };
-
-        let mut visit_count = 0usize;
-
-        {
-            let mut queue_set = FxHashSet::default();
-            let mut queue = BinaryHeap::with_capacity(entries.len());
-            for e in entries.iter().flat_map(|e| e.entries()) {
-                queue.push(NodeWithPriority {
-                    depth: *module_depth.get(&e).unwrap(),
-                    chunk_group_len: 0,
-                    node: ModuleGraph::get_entry(&graphs, e).await?,
-                });
-            }
-            for entry_node in &queue {
-                visitor(
-                    None,
-                    get_node!(graphs, entry_node.node)?,
-                    &mut module_chunk_groups,
-                );
-            }
-            while let Some(NodeWithPriority { node, .. }) = queue.pop() {
-                queue_set.remove(&node);
-                let (node_weight, node) = get_node_idx!(graphs, node)?;
-                let graph = &graphs[node.graph_idx].graph;
-                let neighbors = iter_neighbors_rev(graph, node.node_idx);
-
-                visit_count += 1;
-
-                for (edge, succ) in neighbors {
-                    let succ = GraphNodeIndex {
-                        graph_idx: node.graph_idx,
-                        node_idx: succ,
-                    };
-                    let (succ_weight, succ) = get_node_idx!(graphs, succ)?;
-                    let edge_weight = graph.edge_weight(edge).unwrap();
-                    let action = visitor(
-                        Some((node_weight, edge_weight)),
-                        succ_weight,
-                        &mut module_chunk_groups,
-                    );
-
-                    if action == GraphTraversalAction::Continue && queue_set.insert(succ) {
-                        queue.push(NodeWithPriority {
-                            depth: *module_depth.get(&succ_weight.module).unwrap(),
-                            chunk_group_len: module_chunk_groups
-                                .get(&succ_weight.module)
-                                .unwrap()
-                                .len(),
-                            node: succ,
-                        });
-                    }
-                }
-            }
-        }
+                },
+                |successor, module_chunk_groups| TraversalPriority {
```

Comment:
can you add a comment to explain why this priority is useful?

since the traversal is essentialyl BFS already, is the 'depth' field useful?

Reply:
i guess it is, since it preserves the BFS-ness even though we sort node edges by 'chunk_group_len'